src/Microsoft.DotNet.InternalAbstractions/Native/NativeMethods.Unix.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.DotNet.InternalAbstractions.Native
+{
+    internal static partial class NativeMethods
+    {
+        public static class Unix
+        {
+            public unsafe static string GetUname()
+            {
+                // Utsname shouldn't be larger than 2K
+                var buf = stackalloc byte[2048];
```

Comment:
How do you know this is sufficient space for any platform we might be targeting?  How do you know that padding or header magic or such things doesn't influence where the target field is aligned in the buffer?  This might be fine on the platforms we're currently targeting, but it's risky.